app/decorators/submission_trait_scores_upload_decorator.rb

```diff
@@ -12,14 +12,47 @@ def parser_summary
           group_by{ |id, scores| scores.size }.
           sort{ |score1, score2| score2.first <=> score1.first }
 
-        summary << " - parsed #{trait_scores.size} plant(s) with unique identification"
+        summary << " - parsed #{trait_scores.size} plant scoring unit(s) with unique identification"
         histogram.each do |scoring_number, plants|
-          summary << "  - #{plants.size} plant(s) have #{scoring_number} trait score(s) recorded"
+          summary << "  - #{plants.size} unit(s) have #{scoring_number} trait score(s) recorded"
         end
       end
 
+      accessions = object.submission.content.step04.accessions
```

Comment:
Yes, I read about the other one as well. Involves directly reading the partial file to a string - a strange solution... In such a case we should probably rewrite the controller action to render the partial itself, passing the decorated upload in locales.